### ISSUE TITLE
feat: ML prediction analyst — XGBoost/LightGBM trained on NSE historical data

### ISSUE THREAD

**Issue by ArchieIndian** [hopit-ai/india-trade-cli#145]
## Context
Currently all 7 analyst agents use rule-based Python or LLM reasoning. No quantitative ML predictions. AlphaPy (github.com/ScottfreeLLC/AlphaPy) demonstrates how ML models (XGBoost, LightGBM, CatBoost) can predict price direction, volatility regime, and sector rotation.

### Architecture
```
agent/ml/
  __init__.py
  ml_analyst.py     — MLAnalyst class (implements AnalystBase)
  feature_engine.py — feature computation pipeline
  model_store.py    — model training, saving, loading
  models/           — saved model files (.joblib)
```

### MLAnalyst responsibilities
- Computes features from historical OHLCV data
- Runs trained XGBoost/LightGBM model
- Returns: predicted direction (UP/DOWN/FLAT), probability, confidence
- Feeds into the existing scorecard alongside other analysts

### Feature pipeline (feature_engine.py)
Input: 252-day OHLCV history for a stock

Technical features:
- RSI (14, 21), MACD histogram, Bollinger Band width
- EMA crossover signals (9/21, 20/50)
- ADX, ATR (normalised), volume ratio (vs 20-day avg)
- Price momentum (5d, 10d, 21d returns)

India-specific features:
- FII net flow (today, 5-day momentum)
- VIX level and change
- Days to expiry (weekly, monthly)
- RBI policy proximity (days to next MPC)
- Sector relative strength vs NIFTY

Market regime features:
- Volatility regime (HV20 vs HV60)
- Trend strength (ADX bucket)
- Market breadth (advance/decline ratio)

### Model training (model_store.py)
- Train on 2 years of NSE data
- Target: next-day return > 0 (binary classification)
- Walk-forward validation (train on 18 months, test on 6)
- Models: XGBoost (primary), LightGBM (ensemble), CatBoost (ensemble)
- Retrain monthly or on-demand
- Store models in `~/.trading_platform/ml_models/`

### Integration with multi-agent analysis
```python
class MLAnalyst(AnalystBase):
    def analyze(self, symbol, exchange="NSE"):
        features = self.feature_engine.compute(symbol)
        prediction = self.model.predict_proba(features)
        return AnalystReport(
            analyst="ML Predictor",
            verdict="BULLISH" if prediction > 0.6 else "BEARISH" if prediction < 0.4 else "NEUTRAL",
            confidence=int(abs(prediction - 0.5) * 200),
            data={"prediction_prob": prediction, "features_used": len(features)},
        )
```

### Dependencies
- scikit-learn (already indirect via scipy)
- xgboost
- lightgbm
- joblib (model persistence)

## Priority
Medium — high value feature that differentiates from generic LLM wrappers. The ML predictions provide a quantitative check on the LLM's qualitative reasoning.

## Related
- Inspired by AlphaPy (github.com/ScottfreeLLC/AlphaPy)
- Uses existing data from Fyers/yfinance
- Feeds into existing scorecard system in agent/multi_agent.py
